server/imports/core/connection/index.js

```diff
@@ -170,92 +277,134 @@ Connection.prototype = {
+    const settings = Database.readOne({
+      type: Database.types.Settings,
+      query: {},
+    });
     if (settings.isMigrationDone) return;
 
     const connectionsAfterMigration = [];
 
-    Database.read({ type: Database.types.Connections }).forEach((oldConnection) => {
-      // if there's a name (was mandatory) property it's old.
-      if (oldConnection.name) {
-        let connection = { options: {} };
-        if (oldConnection.url) {
-          connection = this.parseUrl({ url: oldConnection.url });
-          connection.url = oldConnection.url;
-        }
-
-        connection._id = oldConnection._id;
-        connection.connectionName = oldConnection.name;
-
-        migrateSSHPart(oldConnection, connection);
-
-        if (oldConnection.host && oldConnection.port) {
-          connection.servers = [{
-            host: oldConnection.host,
-            port: oldConnection.port,
-          }];
-        }
-        if (oldConnection.readFromSecondary) connection.options.readPreference = 'secondary';
-        else connection.options.readPreference = 'primary';
-
-        if (oldConnection.databaseName) connection.databaseName = oldConnection.databaseName;
-        if (oldConnection.user && oldConnection.password) {
-          connection.scram_sha_1 = {
-            username: oldConnection.user,
-            password: oldConnection.password,
-          };
-          if (oldConnection.authDatabaseName) connection.scram_sha_1.authSource = oldConnection.authDatabaseName;
-          connection.authenticationType = 'scram_sha_1';
+    Database.read({ type: Database.types.Connections }).forEach(
+      (oldConnection) => {
+        // if there's a name (was mandatory) property it's old.
+        if (oldConnection.name) {
+          let connection = { options: {} };
+          if (oldConnection.url) {
+            connection = this.parseUrl({ url: oldConnection.url });
+            connection.url = oldConnection.url;
+          }
+
+          connection._id = oldConnection._id;
+          connection.connectionName = oldConnection.name;
+
+          migrateSSHPart(oldConnection, connection);
+
+          if (oldConnection.host && oldConnection.port) {
+            connection.servers = [
+              {
+                host: oldConnection.host,
+                port: oldConnection.port,
+              },
+            ];
+          }
+          if (oldConnection.readFromSecondary) {
+            connection.options.readPreference = 'secondary';
+          } else connection.options.readPreference = 'primary';
+
+          if (oldConnection.databaseName) {
+            connection.databaseName = oldConnection.databaseName;
+          }
+          if (oldConnection.user && oldConnection.password) {
+            connection.scram_sha_1 = {
+              username: oldConnection.user,
+              password: oldConnection.password,
+            };
+            if (oldConnection.authDatabaseName) {
+              connection.scram_sha_1.authSource =
```

Comment:
Codacy has a fix for the issue: [There should be no line break before or after '='.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
              connection.scram_sha_1.authSource =                oldConnection.authDatabaseName;
```